src/test/java/com/kodilla/ecommercee/domain/carts/CartTestSuite.java

```diff
@@ -0,0 +1,75 @@
+package com.kodilla.ecommercee.domain.carts;
+
+import com.kodilla.ecommercee.domain.products.Product;
+import com.kodilla.ecommercee.repository.CartRepository;
+import com.kodilla.ecommercee.repository.ProductRepository;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class CartTestSuite {
+
+    @Autowired
+    private CartRepository cartRepository;
+    @Autowired
+    private ProductRepository productRepository;
+
+    @Test
+    public void testAddingRowToTheCartEntity() {
+//    Given
+        Cart cart = new Cart();
+
+//    When
+        cartRepository.save(cart);
+        Long cartId = cart.getCartId();
+        List<Cart> carts = (List<Cart>) cartRepository.findAll();
+
+//    Then
+        assertEquals(1, carts.size());
+        assertEquals(carts.get(0).getCartId(), cartId);
+
+//    ClenUp
```

Comment:
cleanup wyrzuciłbym do osobnej metody z annotacją np `@AfterMethod` i tam wywalał wszystkie encje.

Reply:
Sugeruję zmianę w linijce 37 "1" podanej na sztywno na wartość wcześniej pobraną z bazy danych przed dodaniem nowego koszyka. W bazach innych osób pracujących nad projektem mogą już istnieć jakieś testowe koszyki.

Reply:
Raczej nie istnieją bo testy mają czyścić bazę danych po dodaniu wartości testowych. Pozatym test ma sprawdzać konkretny przypadek który jest podany w sekcji given. Plus baza danych zewnętrznych jest zablokowana :)

Reply:
https://kodilla.com/pl/bootcamp-module/146/207/3797#262-code-review-na-czym-polega-dobre-code-review

"Pisząc test jednostkowy nie należy polegać na własnym stanie bazy danych. Co to oznacza? Nie możesz zakładać, że wszyscy w bazie mają takie same dane. Test powinien być uniwersalny - nie możesz zakładać, że po dodaniu do bazy jednego rekordów, będzie on jedynym wpisem w danej tabeli."